_inc/lib/core-api/wpcom-endpoints/publicize-connections.php

```diff
@@ -0,0 +1,163 @@
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_items' ),
+				'permission_callback' => array( $this, 'get_items_permission_check' ),
+			),
+			'schema' => array( $this, 'get_public_item_schema' ),
+		) );
+	}
+
+	/**
+	 * Helper for generating schema. Used by this endpoint and by the
+	 * Connection Test Result endpoint.
+	 *
+	 * @internal
+	 */
+	protected function get_connection_schema_properties() {
+		return array(
+			'id' => array(
+				'description' => __( 'Unique identifier for the Publicize Connection', 'jetpack' ),
+				'type' => 'string',
+			),
+			'service_name' => array(
+				'description' => __( 'Alphanumeric identifier for the Publicize Service', 'jetpack' ),
+				'type' => 'string',
+			),
+			'display_name' => array(
+				'description' => __( 'Username of the connected account', 'jetpack' ),
+				'type' => 'string',
+			),
+			'global' => array(
+				'description' => __( 'Is this connection available to all users?', 'jetpack' ),
+				'type' => 'boolean',
+			),
+		);
+	}
+
+	public function get_item_schema() {
+		$schema = array(
+			'$schema' => 'http://json-schema.org/draft-04/schema#',
+			'title' => 'jetpack-publicize-connection',
+			'type' => 'object',
+			'properties' => $this->get_connection_schema_properties(),
+		);
+
+		return $this->add_additional_fields_schema( $schema );
+	}
+
+	/**
+	 * Helper for retrieving Connections. Used by this endpoint and by
+	 * the Connection Test Result endpoint.
+	 *
+	 * @internal
+	 */
+	protected function get_connections() {
+		global $publicize;
+
+		$items = array();
+
+		foreach ( (array) $publicize->get_services( 'connected' ) as $service_name => $connections ) {
+			foreach ( $connections as $connection ) {
+				$connection_meta = $publicize->get_connection_meta( $connection );
+				$connection_data = $connection_meta['connection_data'];
+
+				$items[] = array(
+					'id' => $publicize->get_connection_unique_id( $connection ),
+					'service_name' => $service_name,
+					'display_name' => $publicize->get_display_name( $service_name, $connection ),
+					'global' => 0 == $connection_data['user_id'],
+				);
+			}
+		}
+
+		return $items;
+	}
+
+	/**
+	 * @param WP_REST_Request $request
+	 * @return WP_REST_Response suitable for 1-page collection
+	 */
+	public function get_items( $request ) {
+		$items = array();
+
+		foreach ( $this->get_connections() as $item ) {
+			$items[] = $this->prepare_item_for_response( $item, $request );
+		}
+
+		$response = rest_ensure_response( $items );
+		$response->header( 'X-WP-Total', count( $items ) );
+		$response->header( 'X-WP-TotalPages', 1 );
+
+		return $response;
+	}
+
+	/**
+	 * Filters out data based on ?_fields= request parameter
+	 *
+	 * @param array $connection
+	 * @param WP_REST_Request $request
+	 * @return array filtered $connection
+	 */
+	function prepare_item_for_response( $connection, $request ) {
```

Comment:
We seem to be lacking a specific visibility modifier here. It'll default to `public`, but since this looks internal, we might want to encapsulate it.

Reply:
It's public: https://core.trac.wordpress.org/browser/trunk/src/wp-includes/rest-api/endpoints/class-wp-rest-controller.php?rev=43571#L186

Fixed in f717306c1.